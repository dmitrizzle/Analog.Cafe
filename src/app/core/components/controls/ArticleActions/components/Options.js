import React from "react"
import styled from "styled-components"

import { GA, makeFroth } from "../../../../../utils"
import { ROUTE_URL_ARTICLES } from "../../../../constants/routes-article"
import { preloadConstructor } from "../../../../utils/routes-article"
import CardCaption from "../../Card/components/CardCaption"
import CardHeader from "../../Card/components/CardHeader"
import CardIntegrated from "../../Card/components/CardIntegrated"
import Link from "../../Link"
import LinkButton from "../../Button/components/LinkButton"
import Placeholder from "../../../vignettes/Picture/components/Placeholder"
import Subscribe from "../../../../../user/components/forms/Subscribe"

export const CardColumns = styled.div`
  display: flex;
  align-items: flex-start;
  ${props => props.theme.size.breakpoint.max.m`
    display: block;
  `};
`
export const CardIntegratedForColumns = styled(CardIntegrated)`
  width: ${props => props.theme.size.breakpoint.stops.min}px;
`
export const CardCaptionIntegrated = styled(CardCaption)`
  font-size: ${props => props.theme.size.font.make.smaller}em !important;
`

export default props => {
  return (
    <CardColumns
      style={{
        display:
          props.user.status !== "ok" || props.nextArticle ? undefined : "block"
      }}
    >
      <div>
        {/* {props.nextArticle && (
          <CardIntegratedForColumns>
            <CardHeader stubborn buttons={[0]} noStar title="Get Published" />
            <CardCaptionIntegrated>
              Do you shoot film? Get your work reviewed and published on
              Analog.Cafe.
            </CardCaptionIntegrated>
            <LinkButton
              branded
              to={props.userStatus === "ok" ? "/submit/compose" : "/submit"}
              onClick={() => {
                GA.event({
                  category: "Campaign",
                  action: "ActionsCard.submit_button"
                })
              }}
              key="Options_LinkButton"
            >
              Write for Analog.Cafe
            </LinkButton>
          </CardIntegratedForColumns>
        )} */}
        <CardIntegratedForColumns>
          <CardHeader stubborn buttons={[0]} noStar title="Email Newsletter" />
          <CardCaptionIntegrated>
            Our weekly email newsletter, “Analogue Reads” is{" "}
            <Link to="https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=8327655f5e">
              excellent
            </Link>
            . <Link to="/privacy-policy">No spam</Link>, no ads. The best way to
            stay in-touch.
          </CardCaptionIntegrated>
          <Subscribe
            subscribeFormCallback={props.subscribeFormCallback}
            stateOverwrite={props.subscribeForm}
            formLocation={"ArticleActions"}
          />
        </CardIntegratedForColumns>
      </div>
      {props.nextArticle &&
        props.nextArticle.slug && (
          <CardIntegratedForColumns>
            <CardHeader
              stubborn
              buttons={[0]}
              noStar
              title={props.nextArticle.title}
              titlePrefix="Next: "
            />
            <figure>
              <Link
                to={ROUTE_URL_ARTICLES + "/" + props.nextArticle.slug}
                onClick={() => {
                  props.nextArticleHeading(
                    preloadConstructor(props.article, props.nextArticle)
                  )
                  GA.event({
                    category: "Navigation",
                    action: "ActionsCard.next_article_picture"
                  })
                }}
              >
                <Placeholder frothId={props.nextArticle.poster}>
                  <img
                    src={
                      makeFroth({ src: props.nextArticle.poster, size: "s" })
                        .src
                    }
                    alt={props.nextArticle.title}
                  />
                </Placeholder>
              </Link>
            </figure>
            <LinkButton
              style={{ margin: 0 }}
              to={ROUTE_URL_ARTICLES + "/" + props.nextArticle.slug}
              onClick={() => {
                props.nextArticleHeading(
                  preloadConstructor(props.article, props.nextArticle)
                )
                GA.event({
                  category: "Navigation",
                  action: "ActionsCard.next_article_button"
                })
              }}
            >
              Read Next <span>➢</span>
            </LinkButton>
          </CardIntegratedForColumns>
        )}
    </CardColumns>
  )
}
