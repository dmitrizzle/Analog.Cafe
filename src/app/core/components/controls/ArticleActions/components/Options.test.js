import { shallow } from "enzyme"
import React from "react"

import Options from "./Options"

it("Render Options without crashing", () => {
  shallow(<Options />)
})
