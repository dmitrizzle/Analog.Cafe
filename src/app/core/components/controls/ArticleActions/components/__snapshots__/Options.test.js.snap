// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Options, matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div>
          <Styled(Styled(styled.div))>
                    <WithTheme(Component)
                              buttons={
                                        Array [
                                                  0,
                                                ]
                              }
                              noStar={true}
                              stubborn={true}
                              title="Get Published"
                    />
                    <Styled(Styled(styled.div))>
                              We’d love to publish your photo essay, guide, or a review!
                    </Styled(Styled(styled.div))>
                    <Unknown
                              branded={true}
                              onClick={[Function]}
                              to="/submit"
                    >
                              Submit Your Article
                    </Unknown>
          </Styled(Styled(styled.div))>
          <Styled(Styled(styled.div))>
                    <WithTheme(Component)
                              buttons={
                                        Array [
                                                  0,
                                                ]
                              }
                              noStar={true}
                              stubborn={true}
                              title="Email Newsletter"
                    />
                    <Styled(Styled(styled.div))>
                              Our weekly email newsletter, “Analogue Reads” is
                               
                              <Unknown
                                        to="https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=8327655f5e"
                              >
                                        excellent
                              </Unknown>
                              . 
                              <Unknown
                                        to="/privacy-policy"
                              >
                                        No spam
                              </Unknown>
                              , no ads. The best way to stay in-touch.
                    </Styled(Styled(styled.div))>
                    <PureComponent
                              formLocation="ArticleActions"
                              stateOverwrite={undefined}
                              subscribeFormCallback={undefined}
                    />
          </Styled(Styled(styled.div))>
</div>,
        <Styled(Styled(styled.div)) />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Styled(Styled(styled.div))>
              <WithTheme(Component)
                            buttons={
                                          Array [
                                                        0,
                                                      ]
                            }
                            noStar={true}
                            stubborn={true}
                            title="Get Published"
              />
              <Styled(Styled(styled.div))>
                            We’d love to publish your photo essay, guide, or a review!
              </Styled(Styled(styled.div))>
              <Unknown
                            branded={true}
                            onClick={[Function]}
                            to="/submit"
              >
                            Submit Your Article
              </Unknown>
</Styled(Styled(styled.div))>,
            <Styled(Styled(styled.div))>
              <WithTheme(Component)
                            buttons={
                                          Array [
                                                        0,
                                                      ]
                            }
                            noStar={true}
                            stubborn={true}
                            title="Email Newsletter"
              />
              <Styled(Styled(styled.div))>
                            Our weekly email newsletter, “Analogue Reads” is
                             
                            <Unknown
                                          to="https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=8327655f5e"
                            >
                                          excellent
                            </Unknown>
                            . 
                            <Unknown
                                          to="/privacy-policy"
                            >
                                          No spam
                            </Unknown>
                            , no ads. The best way to stay in-touch.
              </Styled(Styled(styled.div))>
              <PureComponent
                            formLocation="ArticleActions"
                            stateOverwrite={undefined}
                            subscribeFormCallback={undefined}
              />
</Styled(Styled(styled.div))>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithTheme(Component)
                  buttons={
                                    Array [
                                                      0,
                                                    ]
                  }
                  noStar={true}
                  stubborn={true}
                  title="Get Published"
/>,
                <Styled(Styled(styled.div))>
                  We’d love to publish your photo essay, guide, or a review!
</Styled(Styled(styled.div))>,
                <Unknown
                  branded={true}
                  onClick={[Function]}
                  to="/submit"
>
                  Submit Your Article
</Unknown>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttons": Array [
                    0,
                  ],
                  "noStar": true,
                  "stubborn": true,
                  "title": "Get Published",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "We’d love to publish your photo essay, guide, or a review!",
                },
                "ref": null,
                "rendered": "We’d love to publish your photo essay, guide, or a review!",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "Options_LinkButton",
                "nodeType": "function",
                "props": Object {
                  "branded": true,
                  "children": "Submit Your Article",
                  "onClick": [Function],
                  "to": "/submit",
                },
                "ref": null,
                "rendered": "Submit Your Article",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithTheme(Component)
                  buttons={
                                    Array [
                                                      0,
                                                    ]
                  }
                  noStar={true}
                  stubborn={true}
                  title="Email Newsletter"
/>,
                <Styled(Styled(styled.div))>
                  Our weekly email newsletter, “Analogue Reads” is
                   
                  <Unknown
                                    to="https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=8327655f5e"
                  >
                                    excellent
                  </Unknown>
                  . 
                  <Unknown
                                    to="/privacy-policy"
                  >
                                    No spam
                  </Unknown>
                  , no ads. The best way to stay in-touch.
</Styled(Styled(styled.div))>,
                <PureComponent
                  formLocation="ArticleActions"
                  stateOverwrite={undefined}
                  subscribeFormCallback={undefined}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttons": Array [
                    0,
                  ],
                  "noStar": true,
                  "stubborn": true,
                  "title": "Email Newsletter",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    "Our weekly email newsletter, “Analogue Reads” is",
                    " ",
                    <Unknown
                      to="https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=8327655f5e"
>
                      excellent
</Unknown>,
                    ". ",
                    <Unknown
                      to="/privacy-policy"
>
                      No spam
</Unknown>,
                    ", no ads. The best way to stay in-touch.",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Our weekly email newsletter, “Analogue Reads” is",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "excellent",
                      "to": "https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=8327655f5e",
                    },
                    "ref": null,
                    "rendered": "excellent",
                    "type": [Function],
                  },
                  ". ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": "No spam",
                      "to": "/privacy-policy",
                    },
                    "ref": null,
                    "rendered": "No spam",
                    "type": [Function],
                  },
                  ", no ads. The best way to stay in-touch.",
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "formLocation": "ArticleActions",
                  "stateOverwrite": undefined,
                  "subscribeFormCallback": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div>
            <Styled(Styled(styled.div))>
                        <WithTheme(Component)
                                    buttons={
                                                Array [
                                                            0,
                                                          ]
                                    }
                                    noStar={true}
                                    stubborn={true}
                                    title="Get Published"
                        />
                        <Styled(Styled(styled.div))>
                                    We’d love to publish your photo essay, guide, or a review!
                        </Styled(Styled(styled.div))>
                        <Unknown
                                    branded={true}
                                    onClick={[Function]}
                                    to="/submit"
                        >
                                    Submit Your Article
                        </Unknown>
            </Styled(Styled(styled.div))>
            <Styled(Styled(styled.div))>
                        <WithTheme(Component)
                                    buttons={
                                                Array [
                                                            0,
                                                          ]
                                    }
                                    noStar={true}
                                    stubborn={true}
                                    title="Email Newsletter"
                        />
                        <Styled(Styled(styled.div))>
                                    Our weekly email newsletter, “Analogue Reads” is
                                     
                                    <Unknown
                                                to="https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=8327655f5e"
                                    >
                                                excellent
                                    </Unknown>
                                    . 
                                    <Unknown
                                                to="/privacy-policy"
                                    >
                                                No spam
                                    </Unknown>
                                    , no ads. The best way to stay in-touch.
                        </Styled(Styled(styled.div))>
                        <PureComponent
                                    formLocation="ArticleActions"
                                    stateOverwrite={undefined}
                                    subscribeFormCallback={undefined}
                        />
            </Styled(Styled(styled.div))>
</div>,
          <Styled(Styled(styled.div)) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Styled(Styled(styled.div))>
                <WithTheme(Component)
                                buttons={
                                                Array [
                                                                0,
                                                              ]
                                }
                                noStar={true}
                                stubborn={true}
                                title="Get Published"
                />
                <Styled(Styled(styled.div))>
                                We’d love to publish your photo essay, guide, or a review!
                </Styled(Styled(styled.div))>
                <Unknown
                                branded={true}
                                onClick={[Function]}
                                to="/submit"
                >
                                Submit Your Article
                </Unknown>
</Styled(Styled(styled.div))>,
              <Styled(Styled(styled.div))>
                <WithTheme(Component)
                                buttons={
                                                Array [
                                                                0,
                                                              ]
                                }
                                noStar={true}
                                stubborn={true}
                                title="Email Newsletter"
                />
                <Styled(Styled(styled.div))>
                                Our weekly email newsletter, “Analogue Reads” is
                                 
                                <Unknown
                                                to="https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=8327655f5e"
                                >
                                                excellent
                                </Unknown>
                                . 
                                <Unknown
                                                to="/privacy-policy"
                                >
                                                No spam
                                </Unknown>
                                , no ads. The best way to stay in-touch.
                </Styled(Styled(styled.div))>
                <PureComponent
                                formLocation="ArticleActions"
                                stateOverwrite={undefined}
                                subscribeFormCallback={undefined}
                />
</Styled(Styled(styled.div))>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithTheme(Component)
                    buttons={
                                        Array [
                                                            0,
                                                          ]
                    }
                    noStar={true}
                    stubborn={true}
                    title="Get Published"
/>,
                  <Styled(Styled(styled.div))>
                    We’d love to publish your photo essay, guide, or a review!
</Styled(Styled(styled.div))>,
                  <Unknown
                    branded={true}
                    onClick={[Function]}
                    to="/submit"
>
                    Submit Your Article
</Unknown>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "buttons": Array [
                      0,
                    ],
                    "noStar": true,
                    "stubborn": true,
                    "title": "Get Published",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "We’d love to publish your photo essay, guide, or a review!",
                  },
                  "ref": null,
                  "rendered": "We’d love to publish your photo essay, guide, or a review!",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": "Options_LinkButton",
                  "nodeType": "function",
                  "props": Object {
                    "branded": true,
                    "children": "Submit Your Article",
                    "onClick": [Function],
                    "to": "/submit",
                  },
                  "ref": null,
                  "rendered": "Submit Your Article",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithTheme(Component)
                    buttons={
                                        Array [
                                                            0,
                                                          ]
                    }
                    noStar={true}
                    stubborn={true}
                    title="Email Newsletter"
/>,
                  <Styled(Styled(styled.div))>
                    Our weekly email newsletter, “Analogue Reads” is
                     
                    <Unknown
                                        to="https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=8327655f5e"
                    >
                                        excellent
                    </Unknown>
                    . 
                    <Unknown
                                        to="/privacy-policy"
                    >
                                        No spam
                    </Unknown>
                    , no ads. The best way to stay in-touch.
</Styled(Styled(styled.div))>,
                  <PureComponent
                    formLocation="ArticleActions"
                    stateOverwrite={undefined}
                    subscribeFormCallback={undefined}
/>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "buttons": Array [
                      0,
                    ],
                    "noStar": true,
                    "stubborn": true,
                    "title": "Email Newsletter",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      "Our weekly email newsletter, “Analogue Reads” is",
                      " ",
                      <Unknown
                        to="https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=8327655f5e"
>
                        excellent
</Unknown>,
                      ". ",
                      <Unknown
                        to="/privacy-policy"
>
                        No spam
</Unknown>,
                      ", no ads. The best way to stay in-touch.",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Our weekly email newsletter, “Analogue Reads” is",
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "excellent",
                        "to": "https://us4.campaign-archive.com/?u=256339f7eafa36f2f466aca44&id=8327655f5e",
                      },
                      "ref": null,
                      "rendered": "excellent",
                      "type": [Function],
                    },
                    ". ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": "No spam",
                        "to": "/privacy-policy",
                      },
                      "ref": null,
                      "rendered": "No spam",
                      "type": [Function],
                    },
                    ", no ads. The best way to stay in-touch.",
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "formLocation": "ArticleActions",
                    "stateOverwrite": undefined,
                    "subscribeFormCallback": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
